src/containers/change-resource-confirm-modal/ChangeResourceConfirmModal.tsx

```diff
@@ -26,27 +26,25 @@ const ChangeResourceConfirmModal = ({
   const { t } = useTranslation()
   const { closeModal } = useModalContext()
 
-  const getCoursesAndCooperationsByResourceId = useCallback(
-    () => CoursesAndCooperationsService.getByResourceId(resourceId),
-    [resourceId]
-  )
-
-  const { response, loading } = useAxios({
-    service: getCoursesAndCooperationsByResourceId,
-    defaultResponse: {
-      courses: [],
-      cooperations: []
-    } as CourseCooperationResponse,
-    fetchOnMount: true
+  const { data, isLoading } = useQuery<CourseCooperationResponse>({
+    queryKey: ['getCoursesAndCooperationsByResourceId', resourceId],
+    queryFn: () => CoursesAndCooperationsService.getByResourceId(resourceId),
+    options: {
+      initialData: {
+        courses: [],
+        cooperations: []
+      },
+      enabled: !!resourceId
+    }
   })
 
-  const courses = response.courses.map((course) => ({
+  const courses = data.courses.map((course) => ({
     id: course._id,
     title: course.title,
     subTitle: 'course'
   }))
 
-  const cooperations = response?.cooperations.map((cooperation) => ({
+  const cooperations = data?.cooperations.map((cooperation) => ({
```

Comment:
`data` is defined by `initialData`
```suggestion
  const cooperations = data.cooperations.map((cooperation) => ({
```

Reply:
You are right, thanks. Done.